### PR TITLE
Add option to only show unread messages

### LINKED ISSUE #214: Only show unread messages
Hello @erroneousboat! 

First of all I love this client and really appreciate your nice, clean coding style. 

---

I respect your position of not wanting to create a feature-parity Slack client. However, I think there's one relatively simple feature that a lot of users depend upon to navigate their membership in a large number of channels: Only show unread messages. My org uses Slack _heavily_ and as a result I'm a member of _a lot_ of channels, DMs, and multi-user DMs. By only showing unread messages in the channel list and retaining "Search" functionality, I find that `slack-term` becomes a lot more useful for me. 

I have no expectation that you'll want to add this feature. If you have no desire to add it, I'm happy to continue maintaining my own fork, as it's the only way that I can reasonably use `slack-term` at my org. However, I wanted to propose this addition if you think it might be helpful to other users as well. 

PR: https://github.com/erroneousboat/slack-term/pull/215/

Cheers,

---

Version: master 
Installation method: build

### PR DESCRIPTION
**Pull request by acaloiaro**
This PR adds support for a new config option `"show_unread_only": true` that results in only unread messsages being shown in the channel list. 

Fixes https://github.com/erroneousboat/slack-term/issues/214

### CODE REVIEW

File: components/channels.go [hunk 1]
```diff
@@ -97,41 +97,76 @@ func (c ChannelItem) GetChannelName() string {
...
-	CursorPosition  int // the y position of the 'cursor'
-
-	SearchMatches  []int // index of the search matches
-	SearchPosition int   // current position of a search match
+	ChannelItems    map[string]*ChannelItem // mapping of channel IDs to ChannelItems
+	CursorPosition  int                     // the y position of the cursor in this component
+	List            *termui.List            // ui of visible channels
+	Offset          int                     // offset in channel list from which items are render
+	SearchPosition  int                     // current position in search results
+	SelectedChannel string                  // index of which channel is selected from the List
```
Comment (acaloiaro) [inline components/channels.go]: Update comment: `ID of currently selected channel`

**Comment by j6k4m8** [comment]
Just wanted to +1 this, and also offer to help resolve merge conflicts if that's helpful to you, @acaloiaro!

**Comment by acaloiaro** [comment]
Hi @j6k4m8 -- if you want to bring this PR across the finish line, you may want to cherry-pick the following commits from my repo:

https://github.com/acaloiaro/slack-term/commit/fd4f950cc6f7a43818836860b73b558833b137cf
https://github.com/acaloiaro/slack-term/commit/85e4f333e7eaa2dab12e97a28a253f3876592ccc

This PR has been open on this repo for quite a while without any attention, so I don't have any plans to advocate for it here. 

Feel free to tak the reins! Cheers.

**Comment by j6k4m8** [comment]
Might just use yours then, @acaloiaro — didn't realize you had a live fork :)

**Comment by el-iot** [comment]
I think this would be a great feature-add and frankly it would make slack-term usable for me (my workspace also uses a lot of channels and when I use Slack I normally hide all conversations that don't have notifications for me).

**Pull request closed without merging**